### DISCUSSION TITLE
Unable to use CompiledBinding in MultiBinding.

### DISCUSSION THREAD

**Discussion by kyuranger** [AvaloniaUI/Avalonia#13045] [Q&A] [answered]
This is a simple example of a simplified code from my original program. If CompiledBinding is not turned on, it can compile and run, but if CompiledBinding is turned on, it prompts 

> Unable to resolve property or method of name 'Source' on type 'Avalonia.Data.MultiBinding'

It seems because Source is a property of Image cause this.

Here is the XAML:
```
<Window xmlns="https://github.com/avaloniaui"
        xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
        xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
        xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
        mc:Ignorable="d" d:DesignWidth="800" d:DesignHeight="450"
        x:Class="AvaloniaApplication2.MainWindow"
		xmlns:Local="using:AvaloniaApplication2"
        Title="AvaloniaApplication2">
	<Window.Resources>
		<Local:TestConverter x:Key="TestConverter"></Local:TestConverter>
	</Window.Resources>
	<Image Height="100" VerticalAlignment="Center" HorizontalAlignment="Center" Width="100">
		<Image.Height>
			<MultiBinding Converter="{StaticResource TestConverter}">
				<Binding Path="$parent.Bounds.Height"></Binding>
				<Binding Path="$self.Source.Size"></Binding>
			</MultiBinding>
		</Image.Height>
	</Image>
</Window>
```

And here is code-behind:
```
using Avalonia.Controls;
using Avalonia.Data.Converters;
using System;
using System.Globalization;

namespace AvaloniaApplication2
{
    public partial class MainWindow : Window
    {
        public MainWindow()
        {
            InitializeComponent();
        }
    }
    public class TestConverter : IMultiValueConverter
   {
     public object? Convert(IList<object?> values, Type targetType, object? parameter, CultureInfo culture)
     {
         return values[0];
     }
    }
}
```

What's wrong with my code?

**Comment by IanRawley** [comment]
My initial instinct is that Multibindings don't use IValueConverter, they use IMultiValueConverter.  So I can see why compilation would fail there.  But I'm not sure why reflection bindings wouldn't.

**Reply by kyuranger** [reply]
Actually it is an IMultiValueConverter, I have fixed this error in the question, but the error still exists.

**Reply by kyuranger** [reply]
According to CompiledBinding tutorial, it seems there are no limitation of Multibindings

**Reply by IanRawley** [reply, marked as answer]
Looking at the error message, it seems the $self binding is pointing at the multi-binding rather than the image.  If you make it more explicitly the image does it work?

**Reply by kyuranger** [reply]
Well, that's the problem. After I set the name of the Image to I and change the Path to `#I.Source.Size`. Now it works.
